### PR TITLE
fix: added component-test-setup to gamut-kit

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### Overview

Adds `component-test-setup` as dependency to `@codecademy/gamut-kit` so consuming apps don't need to specify it.

### CODE REVIEW

**Review by codecaaron — APPROVED** [review]

**Review by dreamwasp — APPROVED** [review]

**Review by MattSchiller — APPROVED** [review]
Oooo yeahhhh! So once this is merged, we'd want to remove it from the deps in the monolith et al, right? Loving the closing of this (minor) pitfall.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.16-alpha.d7090a.0<br/>@codecademy/gamut-labs@13.4.8-alpha.d7090a.0<br/>@codecademy/gamut-tests@2.3.9-alpha.d7090a.0<br/>@codecademy/gamut@29.9.8-alpha.d7090a.0<br/>@codecademy/konamimojisplosion@0.2.9-alpha.d7090a.0<br/>@codecademy/styleguide@33.9.8-alpha.d7090a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60761e55520aa200d31f1d67--gamut-preview.netlify.app>https://60761e55520aa200d31f1d67--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60761e55520aa200d31f1d67>Deploy Logs</a></p>

File: packages/gamut/src/AccordionArea/__tests__/AccordionArea-test.tsx [hunk 1]
```diff
@@ -40,9 +40,7 @@ describe('AccordionArea', () => {
     await act(async () => {
       jest.runAllTimers();
     });
-    // @ts-expect-error - component-test-setup has a bug right now where wrapper is of type ReactWrapper<unknown>
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/AccordionArea/__tests__/AccordionArea-test.tsx:43]: gosh I love this feature

**Pull request merged**